client/signup/steps/plans/util.ts

```diff
@@ -0,0 +1,130 @@
+import {
+	planHasFeature,
+	FEATURE_UPLOAD_THEMES_PLUGINS,
+	isEcommerce,
+} from '@automattic/calypso-products';
+import { getUrlParts } from '@automattic/calypso-url';
+import { MinimalRequestCartProduct } from '@automattic/shopping-cart';
+import wp from 'calypso/lib/wp';
+
+type DomainItem = {
+	is_domain_registration?: boolean;
+	meta?: string;
+	product_slug?: string;
+};
+
+type PlanProperties = {
+	additionalStepData?: object;
+	flowName: string;
+	launchSite?: boolean;
+	selectedSite?: { ID: number; name: string } | null;
+	stepName: string;
+	stepSectionName: string | undefined;
+	themeSlugWithRepo?: string;
+	errorNotice: ( message: string ) => void;
+	goToNextStep: () => void;
+	recordTracksEvent: ( eventName: string, eventProperties: object ) => void;
+	submitSignupStep: ( step: object, providedDependencies: object ) => void;
+};
+
+export const getIntervalType = (): string => {
+	const urlParts = getUrlParts( typeof window !== 'undefined' ? window.location?.href : '' );
+	const intervalType = urlParts?.searchParams.get( 'intervalType' ) || '';
+
+	if ( [ 'yearly', 'monthly' ].includes( intervalType ) ) {
+		return intervalType;
+	}
+
+	// Default value
+	return 'yearly';
+};
+
+export const getDomainName = ( domainItem: DomainItem ): string | undefined => {
+	return domainItem && domainItem.meta;
+};
+
+export const buildUpgradeFunction = (
```

Comment:
If we are moving this function out of the component it should ideally live in step-actions. This is sort of where the Async Api Calls live in relation to Steps.

https://github.com/Automattic/wp-calypso/blob/cf7823160cc6baf87f4631051b0127dcd44971d2/client/lib/signup/step-actions.js